Historical state reindex for trusted node sync

arnetheduck opened the pull request:
When performing trusted node sync, historical access is limited to
states after the checkpoint.

Reindexing restores full historical access by replaying historical
blocks against the state and storing snapshots in the database.

The process can be initiated or resumed at any point in time.

arnetheduck committed:
work around test gc issue

arnetheduck committed:
buggy bounds check

arnetheduck committed:
fix seq size

arnetheduck committed:
comments

zah commented on beacon_chain/consensus_object_pools/blockchain_dag.nim:
```diff
@@ -1770,3 +1781,142 @@ proc aggregateAll*(
+  dag.tail = dag.genesis
+
+  if junk.len > 0:
+    info "Dropping redundant states", junk
```
Wouldn't this print too much data?

arnetheduck replied:
typically no (junk gets created during a crash only, really - but ..)

arnetheduck replied:
f3df03d138171135cbec0d14068268c7e819f87e

arnetheduck committed:
Historical state reindex for trusted node sync

When performing trusted node sync, historical access is limited to
states after the checkpoint.

Reindexing restores full historical access by replaying historical
blocks against the state and storing snapshots in the database.

The process can be initiated or resumed at any point in time.

arnetheduck committed:
shorten log

arnetheduck committed:
comments
